core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -1292,8 +1290,7 @@ object KafkaConfig {
       .define(SaslLoginRefreshMinPeriodSecondsProp, SHORT, Defaults.SaslLoginRefreshMinPeriodSeconds, MEDIUM, SaslLoginRefreshMinPeriodSecondsDoc)
       .define(SaslLoginRefreshBufferSecondsProp, SHORT, Defaults.SaslLoginRefreshBufferSeconds, MEDIUM, SaslLoginRefreshBufferSecondsDoc)
       /** ********* Delegation Token Configuration ****************/
-      .define(DelegationTokenSecretKeyAliasProp, PASSWORD, null, MEDIUM, DelegationTokenSecretKeyAliasDoc)
-      .define(DelegationTokenSecretKeyProp, PASSWORD, null, MEDIUM, DelegationTokenSecretKeyDoc)
+      .define(DelegationTokenSecretKeyProp, PASSWORD, MEDIUM, DelegationTokenSecretKeyDoc)
```

Comment:
It looks like some logic depends on having `null` as default as a bunch of tests are failing with:
```
org.apache.kafka.common.config.ConfigException: Missing required configuration "delegation.token.secret.key" which has no default value.
```

Reply:
Will fix if we agree that this is a good idea. :) It's more aggressive than I would usually recommend, so I'm looking for additional options.

Reply:
@mimaison so, what's your take? Should we remove this for 3.0, or keep it until 4.0?